fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -303,55 +309,87 @@ public DataApiRequest(
     }
 
     /**
-     * Update sort field string by truncating the dateTime if it is present.
+     * To check whether dateTime column request is first one in the sort list or not.
      *
-     * @param sorts  String of sort columns
+     * @param sortColumns  map of columns and its direction
      *
-     * @return Updated sort column if dateTime exists. Otherwise return as it is received.
+     * @return True if dateTime column is first one in the sort list. False otherwise
      */
-    protected String truncateTimeSort(String sorts) {
+    protected Boolean isDateTimeFirstSortField(Map<String, SortDirection> sortColumns) {
+        if (sortColumns != null) {
+            List<String> columns = new ArrayList(sortColumns.keySet());
```

Comment:
I have updated the contract to make sure argument type is LinkedHashMap. Will commit the changes after addressing all the review comments.